ci: switch to google-github-actions/auth for GKE based workflows

tklauser opened the pull request:
service_account_key in the google-github-actions/setup-gcloud action is
deprecated, see [1], [2]. Switch to the suggested
google-github-actions/auth action instead.

[1] https://github.com/cilium/cilium/actions/runs/2997569300
[2] https://github.com/google-github-actions/setup-gcloud#authorization

In the multicluster workflows, move the checkout step before the
credentials setup so authentication is shared with future steps of the
job.

sayboras reviewed (approved):
Looks good to me, as long as CIs are happy :pray:

tklauser committed:
ci: switch to google-github-actions/auth for GKE based workflows

service_account_key in the google-github-actions/setup-gcloud action is
deprecated, see [1], [2]. Switch to the suggested
google-github-actions/auth action instead.

[1] https://github.com/cilium/cilium/actions/runs/2997569300
[2] https://github.com/google-github-actions/setup-gcloud#authorization

In the GKE and multicluster workflows, move the checkout step before the
credentials setup so authentication is shared with future steps of the
job.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

nbusseneau reviewed (approved):
Thanks. Ultimately it would be better if we switched to OIDC for all providers, as we already did for Azure, but I don't have time for it right now. This will help when the time comes though, won't have to rework the whole structure :p
